build01/marvin-mk2.nix

```
@@ -0,0 +1,65 @@
+{ pkgs, lib, config, ... }:
+let
+  userLib = import ../users/lib.nix { inherit lib; };
+
+  sources = import ../nix/sources.nix;
+  marvin-mk2 = pkgs.python3.pkgs.buildPythonApplication rec {
+    pname = "marvin-mk2";
+    version = "rolling";
+    src = import sources.marvin-mk2 { };
+
+    propagatedBuildInputs = with pkgs.python3.pkgs; [
+      aiohttp
+      gidgethub
+    ];
+  };
+
+  marvin-mk2-runtimeenv = with pkgs; [
+    marvin-mk2
+  ];
+
+in
+{
+  users.users.marvin-mk2.packages = [
+  ];
+  users.groups.marvin-mk2 = { };
+  users.users.marvin-mk2 = {
+    useDefaultShell = true;
+    uid = userLib.mkUid "mmk2";
+    extraGroups = [ "marvin-mk2" ];
+  };
+  nix.trustedUsers = [
+    "marvin-mk2"
+  ];
+
+  systemd.services.marvin-mk2 = {
+    description = "marvin-mk2 service";
+    enable = true;
+    path = marvin-mk2-runtimeenv;
+    environment.XDG_CONFIG_HOME = "/var/lib/marvin-mk2";
+    environment.XDG_CACHE_HOME = "/var/cache/marvin-mk2";
```

small nitpick, you can use:
```suggestion
    environment.XDG_CACHE_HOME = "%C";
```

See https://www.freedesktop.org/software/systemd/man/systemd.unit.html#Specifiers

---

I'm not sure why/if its even necessary to set these in the first place. What do they default to? Currently marvin-mk2 doesn't need any cache/config anyways, but that may change.

(again, the settings are included since I based this of r-ryantm)